### PR TITLE
Populate the docs with the Mu code release process.

### PR DESCRIPTION
**Pull request by corthon**

### CODE REVIEW

File: docs/How/release_process.md [hunk 1]
```diff
@@ -1,2 +1,88 @@
...
+
+In general, the life-cycle of active code follows the following path:
+
+<center>![Stable Release, Upstream Rebase, Build/Boot Fixes, Active Development, Stabilization, Stable Release](../img/release_cycle.png)</center>
+
+All active work in Project Mu is performed on a `release/*` branch, usually named sequentially according to the date of TianoCore commit that it's based on (eg. `release/201808` is based on the `edk2-stable201808` branch in TianoCore). Work proceeds on that branch until a new TianoCore integration is targeted, at which point a new branch in created and all existing changes are rebased onto the new branch and the new branch is used for all active development going forward. At this point, the previous branch enters a stabilization period where further tests are performed and only bug fixes are allowed to be committed. After stabilization, the branch is labeled as `stable` and will only receive critical bug fixes either directly to the branch or backported from a more recent release.
```
Comment (out0xb2) [inline docs/How/release_process.md]: How about we remove "usually".  If we later find an exception, we can document that.
Also, "eg." should have 2 periods "e.g."

File: docs/How/release_process.md [hunk 2]
```diff
@@ -1,2 +1,88 @@
...
+
+`release/*` branches will be maintained in LTS for at least the next two releases.
```
Comment (out0xb2) [inline docs/How/release_process.md]: LTS is not yet defined (defined below)

File: docs/How/release_process.md [hunk 3]
```diff
@@ -1,2 +1,88 @@
...
+* A change was reverted or modified more recently in Mu history, and the history of this change was squashed to maintain simplicity when comparing with upstream (TianoCore).
+
+Once all evaluation is completed, the rebase will be performed in the new `release/*` branch. This branch will then be built for a reference platform (to be selected by internal team) and booted, at which point it will be considered the active development branch.
+
+### Integration Milestone Tags
+
+During integration, multiple tags are applied to the branch to serve as milestones. They also serve as reference point for changelog documentation that is produced during the integration process. These tags are described below:
+
+* `*_Upstream`
+    * This tag is placed on the exact TianoCore commit that a given release branch started from. This is used as a reference point between branches and relative to the rebase operation. The documentation produced for this tag contains the differences in TianoCore between this branch and the previous branch. For branches that originated from TianoCore releases, this changelog should be almost identical to the TianoCore changelog.
```
Comment (out0xb2) [inline docs/How/release_process.md]: Should we explain "almost"?

File: docs/How/release_process.md [hunk 4]
```diff
@@ -1,2 +1,88 @@
...
+
+### Public Contribution/Commentary
+
+For information on the contribution policies and steps, see the [How to Contribute](/How/contributing) document.
+
+### Upstream Cherry-Picks
+
+In the event that a critical change is made in the TianoCore upstream during the Active Development phase, the Project Mu team (with any suggestions or comment from downstream contributors) will evaluate the change for a mid-release cherry pick. If warranted, the commit(s) will be cherry-picked directly from TianoCore and prefixed with a "CHERRY-PICK" tag in the commit message to they can be cleaned up in the next rebase.
```
Comment (out0xb2) [inline docs/How/release_process.md]: s/to they/so they/

```suggestion
In the event that a critical change is made in the TianoCore upstream during the Active Development phase, the Project Mu team (with any suggestions or comment from downstream contributors) will evaluate the change for a mid-release cherry pick. If warranted, the commit(s) will be cherry-picked directly from TianoCore and prefixed with a "CHERRY-PICK" tag in the commit message so they can be cleaned up in the next rebase.
```

File: docs/How/release_process.md [hunk 5]
```diff
@@ -1,2 +1,88 @@
...
+## Overview
+
+!!! warning "Contents and Process Under Active Development"
+    The basics of this process are identical to those followed by the Project Mu firmware integration and release process internal to Microsoft, but the formal documentation, branch naming, and tagging process is a work in progress. While this is how we expect things to work, there may be changes within the the first few releases driven by feedback within the team and any external consumers/contributors.
+
+In the interest of maintaining a close, well-defined relationship with the upstream project, TianoCore, the active release branch of Project Mu is periodically deprecated and all Mu-related changes are rebased onto a selected commit of TianoCore. This keeps Project Mu up to date with TianoCore while highlighting all Project Mu differences in the most recent commits and encouraging the reverse integration of all universal changes/fixes back into TianoCore.
+
+In general, the life-cycle of active code follows the following path:
+
+<center>![Stable Release, Upstream Rebase, Build/Boot Fixes, Active Development, Stabilization, Stable Release](../img/release_cycle.png)</center>
```
Comment (out0xb2) [inline docs/How/release_process.md]: Can we update this to exactly match the headings below or update the headings below to match this diagram?
Is it "Upstream Rebase" or "Upstream Integration Phase"?

File: docs/How/release_process.md [hunk 6]
```diff
@@ -1,2 +1,88 @@
-# Overview
-
+## Overview
+
+!!! warning "Contents and Process Under Active Development"
+    The basics of this process are identical to those followed by the Project Mu firmware integration and release process internal to Microsoft, but the formal documentation, branch naming, and tagging process is a work in progress. While this is how we expect things to work, there may be changes within the the first few releases driven by feedback within the team and any external consumers/contributors.
+
+In the interest of maintaining a close, well-defined relationship with the upstream project, TianoCore, the active release branch of Project Mu is periodically deprecated and all Mu-related changes are rebased onto a selected commit of TianoCore. This keeps Project Mu up to date with TianoCore while highlighting all Project Mu differences in the most recent commits and encouraging the reverse integration of all universal changes/fixes back into TianoCore.
```
[the remark this answers is not in the document]
Comment (corthon) [reply docs/How/release_process.md]: Fair point. Would "... encouraging the reverse integration of all changes/fixes as soon as they are ready for upstream." be better?

File: docs/How/release_process.md (hunk 3 above)
Comment (corthon) [reply docs/How/release_process.md]: Actually, now that I read it again, I think we should just ditch the "almost".

File: docs/How/release_process.md (hunk 5 above)
Comment (corthon) [reply docs/How/release_process.md]: You know, Sean said the same thing and I just forgot. Will do (to the extent possible).

Comment (corthon) [reply docs/How/release_process.md]: Wait, apparently I was thinking of something else. Errr... graphics are a pain...

Comment (corthon) [reply docs/How/release_process.md]: Done and done. If you're gonna have one drawing, might as well commit to drawing. All these graphics need to be redone later, anyway.

File: docs/How/release_process.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (out0xb2) [reply docs/How/release_process.md]: It works... but I sort of like us setting the bar that Mu changes should only be of a class and quality that we believe should be ready to upstream.

File: docs/How/release_process.md [hunk 7]
```diff
@@ -1,2 +1,88 @@
...
+
+In general, the life-cycle of active code follows the following path:
+
+<center>![Stable Release, Upstream Rebase, Build/Boot Fixes, Active Development, Stabilization, Stable Release](../img/release_cycle.png)</center>
+
+All active work in Project Mu is performed on a `release/*` branch, named sequentially according to the date of TianoCore commit that it's based on (e.g. `release/201808` is based on the `edk2-stable201808` branch in TianoCore). Work proceeds on that branch until a new TianoCore integration is targeted, at which point a new branch in created and all existing changes are rebased onto the new branch and the new branch is used for all active development going forward. At this point, the previous branch enters a stabilization period where further tests are performed and only bug fixes are allowed to be committed. After stabilization, the branch is labeled as `stable` and will only receive critical bug fixes either directly to the branch or backported from a more recent release.
```
[the remark this answers is not in the document]
Comment (corthon) [reply docs/How/release_process.md]: Done.

File: docs/How/release_process.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (corthon) [reply docs/How/release_process.md]: Settled offline.

File: docs/How/release_process.md [hunk 8]
```diff
@@ -9,7 +9,7 @@ In general, the life-cycle of active code follows the following path:
...
+All active work in Project Mu is performed on a `release/*` branch, named sequentially according to the date of TianoCore commit that it's based on (e.g. `release/201808` is based on the `edk2-stable201808` branch in TianoCore). Work proceeds on that branch until a new TianoCore integration is targeted, at which point a new branch is created and all existing changes are rebased onto the new branch and the new branch is used for all active development going forward. At this point, the previous branch enters a stabilization period where further tests are performed and only bug fixes are allowed to be committed. After stabilization, the branch is labeled as `stable` and will only receive critical bug fixes either directly to the branch or backported from a more recent release.
```
Comment (out0xb2) [inline docs/How/release_process.md]: Ignore this comment

**Pull request merged**